README.ConfigureResourceReferences.md

```diff
@@ -0,0 +1,185 @@
+
+After you have a basic understanding of the fields for the target resource, some
+approaches would help you identify whether the field is mapping to a GCP
+resource.
+
+1.  In the GCP tutorial for the target resource, if another GCP resource is
+    required before creating the target resource, and if the value of a field in
+    the other GCP resource is needed when creating the target resource, then we
+    know the field is likely a reference field. Note down the field name and the
+    other GCP resource name.
+1.  In the TF sample for the target resource, if the value of a field is an
+    attribute reference of another TF resource, then we know the field is likely
+    a reference field. Note down the field name and the other TF resource type
+    name. E.g. the `key_ring` fields mentioned in the
+    [cmek sample for PubSubTopic](https://registry.terraform.io/providers/hashicorp/google-beta/latest/docs/resources/pubsub_topic#example-usage---pubsub-topic-cmek).
+1.  Be aware of field name mentioning: `name`, `id`, `link`, `email`,
+    `service_account`, resource names under the same API of the target resource,
+    or field description mentioning `refer`, `GCP`/`google`/`cloud` resources,
+    `url`, `uri`, etc. Once you see these keywords, it's very likely that this
+    field references to gcp resource. Double-check the GCP tutorial and/or TF
+    documentation to confirm.
+
+Figure out the referenced GCP resource through reading the field description. If
+you are still unsure, contact the API team for more information. Once it's
+determined, note down the field name and the referenced GCP resource name.
+
+## Identify whether the referenced resource is TF-based or DCL-based
+
+### Turn GCP resource name or TF type name into KCC kind name
+
+Before moving on, you need to turn the referenced TF type/GCP resource names
+you've noted down into KCC kind names. Ask the KCC team for help if you still
+can't figure them out with the following instructions.
+
+1.  If you know the TF type name, you can do project-scoped search in your local
+    repository with the TF type name. If you find a hit in a service mapping
+    yaml file (under either the `config/servicemappings/` folder or the
+    `scripts/resource-autogen/generated/servicemappings/`), the corresponding
+    `kind` field in the same resource config should be the KCC kind.
+
+1.  If you know the GCP resource name, you can either use the search widget to
+    find the corresponding TF type in
+    [Terraform doc](https://registry.terraform.io/providers/hashicorp/google-beta/latest/docs)
+    with the GCP resource name, then figure out the KCC kind name following the
+    step above; or you can use the search widget to find the corresponding KCC
+    kind directly in
+    [KCC reference doc](https://cloud.google.com/config-connector/docs/reference/overview).
+
+1.  If you can't find a KCC kind with the steps above, it's possible that the
```

Comment:
Similarly instead of saying "Reach out to the KCC team", let's just ask the contributor to raise a question as a GitHub issue. Or we simply remove this sentence.